pkg/platform/runtime/buildexpression/buildexpression.go

```diff
@@ -906,7 +910,17 @@ func (v *Value) MarshalJSON() ([]byte, error) {
 	case v.Ap != nil:
 		return json.Marshal(v.Ap)
 	case v.List != nil:
-		return json.Marshal(v.List)
+		// Buildexpression list order does not matter, so sorting is necessary for
+		// comparisons. Go's JSON marshaling is deterministic, so utilize that.
+		// This should not be necessary when PB-4607 is implemented.
+		list := make([]*Value, len(*v.List))
+		copy(list, *v.List)
+		sort.SliceStable(list, func(i, j int) bool {
+			b1, err1 := json.Marshal(list[i])
+			b2, err2 := json.Marshal(list[j])
+			return err1 == nil && err2 == nil && string(b1) < string(b2)
+		})
+		return json.Marshal(list)
```

Comment:
This is the kicker. When marshaling buildexpression's to JSON, lists need to be sorted since order doesn't matter. That way we can perform comparisons for equality.

The rest of this PR you've seen and approved.